addons/phone_validation/models/phone_blacklist.py

```diff
@@ -40,15 +45,19 @@ def create(self, values):
             done.add(sanitized)
             to_create.append(dict(value, number=sanitized))
 
-        """ To avoid crash during import due to unique email, return the existing records if any """
-        sql = '''SELECT number, id FROM phone_blacklist WHERE number = ANY(%s)'''
-        numbers = [v['number'] for v in to_create]
-        self._cr.execute(sql, (numbers,))
-        bl_entries = dict(self._cr.fetchall())
-        to_create = [v for v in to_create if v['number'] not in bl_entries]
+        # Get existing phone blacklist (for requested numbers to create)
+        numbers_requested = [tc['number'] for tc in to_create]
+        pb_existing = self.search([('active', 'in', [True, False]), ('number', 'in', numbers_requested)])
 
-        results = super(PhoneBlackList, self).create(to_create)
-        return self.env['phone.blacklist'].browse(bl_entries.values()) | results
+        # Out of existing pb records, activate non-active
+        pb_existing.filtered(lambda pb: not pb.active).write({'active': True})
+
+        # Create new records, while skipping existing_numbers
+        existing_numbers = pb_existing.mapped('number')
+        to_create_filtered = [d for d in to_create if d['number'] not in existing_numbers]
+        pb_created = super().create(to_create_filtered)
+
+        return pb_existing | pb_created
```

Comment:
Could be great to keep ordering between input and output (normally output should follow same order as input, so that we could zip them)

Reply:
Not sure I understand this one. I guess that you want to return results in the order given in the values arg, right ?

so assuming we'd call
`create([{'111 111 111'},{'222 222 222'}, {'333 333 333'}])`

but it happens that number `'222 222 222'` already exist

we'd want to get results from create ordered like so:
`<PhoneBlackList ...'111 111 111'>, <PhoneBlackList ...'222 222 222'>, <PhoneBlackList ..'333 333 333'>`
instead of (what we had in my current code):
we'd want to get results from create ordered like so:
`<PhoneBlackList ...'222 222 222'>, <PhoneBlackList ...'111 111 111'>, <PhoneBlackList ..'333 333 333'>`

Reply:
Indeed: ordering of returned recordset should match ordering of incoming values.